### PR TITLE
Dan/2023/03/add a check registration step

### PR DESCRIPTION
**Pull request by DJViau**
Same as [this PR](https://github.com/ProjectOpenSea/seaport/pull/1063/files) (closed in favor of this one), but it also adds a check registration step.

### CODE REVIEW

File: test/foundry/new/FuzzEngine.t.sol [hunk 1]
```diff
@@ -17,6 +17,7 @@ import {
     HashValidationZoneOfferer
 } from "../../../contracts/test/HashValidationZoneOfferer.sol";
 import { AdvancedOrder, FuzzHelpers } from "./helpers/FuzzHelpers.sol";
+import { CheckHelpers } from "./helpers/FuzzSetup.sol";
```
Comment (horsefacts) [inline test/foundry/new/FuzzEngine.t.sol:20]: Might make sense to move this to `FuzzEngineLib`, or `FuzzTestContextLib`, what do you think?

File: test/foundry/new/FuzzEngine.t.sol [hunk 2]
```diff
@@ -493,6 +492,9 @@ contract FuzzEngineTest is FuzzEngine {
                 orders[0].toBasicOrderParameters(orders[0].getBasicOrderType())
             );
 
+        context.registerCheck(this.check_orderFulfilled.selector);
+        context.registerCheck(this.check_orderStatusFullyFilled.selector);
```
Comment (horsefacts) [inline test/foundry/new/FuzzEngine.t.sol:496]: `registerCheck` returns the `TestContext`, so these are chainable if you want, e.g.:

```
context.registerCheck(this.check_orderFulfilled.selector).registerCheck(this.check_orderStatusFullyFilled.selector);
```

**Review by horsefacts — APPROVED** [review]

File: test/foundry/new/FuzzEngine.t.sol (hunk 2 above)
Comment (DJViau) [reply test/foundry/new/FuzzEngine.t.sol:496]: I forgot that.  I just changed it and prettier makes it uglier.  imo it's more readable as-is, but I'm happy to chain if we want to adopt chain-when-possible or something like that as a convention.

File: test/foundry/new/FuzzEngine.t.sol (hunk 1 above)
Comment (DJViau) [reply test/foundry/new/FuzzEngine.t.sol:20]: Yeah, sg.  FuzzEngineLib seems like he right play based on gut feel.  Happy to reshuffle as necessary.  Will update and then merge into your branch

**Pull request merged**